internal/proxy/oauthproxy.go

```diff
@@ -802,6 +802,9 @@ func (p *OAuthProxy) Authenticate(rw http.ResponseWriter, req *http.Request) (er
 	req.Header.Set("X-Forwarded-Email", session.Email)
 	req.Header.Set("X-Forwarded-Groups", strings.Join(session.Groups, ","))
 
+	for key, val := range p.upstreamConfig.InjectRequestHeaders {
```

Comment:
Let's move these above setting of the `X-Forwarded-` headers. We don't want to allow impersonation by setting these headers from a config.

Reply:
moved